templates/marathon.upstart.erb

```diff
@@ -6,4 +6,16 @@ stop on runlevel [!2345]
 respawn
 respawn limit 10 5
 
-exec su -c '<%= scope.lookupvar('marathon::bin_path') %>/marathon --master <%= scope.lookupvar('marathon::master') %> --zk <%= scope.lookupvar('marathon::zk') %> <%= scope.lookupvar('marathon::extra_options') %>' - <%= scope.lookupvar('marathon::user') %>
+setuid <%= scope.lookupvar('marathon::user') %>
+
+exec <%= scope.lookupvar('marathon::bin_path') %>/marathon \
+--master <%= scope.lookupvar('marathon::master') %> \
+--zk <%= scope.lookupvar('marathon::zk') %> \
+<%- if scope.lookupvar('marathon::authenticate') -%>
+    --mesos_authentication_principal <%= scope.lookupvar('marathon::auth_principal') %> \
+    --mesos_authentication_secret_file <%= scope.lookupvar('marathon::secret_file') %> \
+    <%- if scope.lookupvar('marathon::auth_role') -%>
+        --mesos_role <%= scope.lookupvar('marathon::auth_role') %> \
```

Comment:
@EvanKrall you can use ``if scope.lookupvar('marathon::auth_role') == :undef`` . This is the recommended way from puppetlabs http://docs.puppetlabs.com/guides/templating.html#testing-for-undefined-variables